### PR TITLE
Do not shadow the class member return_type [blocks: #2310]

### PR DESCRIPTION
**Pull request by tautschnig**
Using the very short variable name "r" to make clear this is only of very local
use instead.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/ansi-c/c_typecheck_expr.cpp [hunk 1]
```diff
@@ -927,10 +927,10 @@ void c_typecheck_baset::typecheck_side_effect_statement_expression(
 
     code_function_callt &fc=to_code_function_call(last);
 
-    const auto &return_type = to_code_type(fc.function().type()).return_type();
+    const auto &r = to_code_type(fc.function().type()).return_type();
```
Comment (peterschrammel) [inline src/ansi-c/c_typecheck_expr.cpp]: Suggestions: `return_type_ref`, `call_return_type` ?

File: src/ansi-c/c_typecheck_expr.cpp [hunk 2]
```diff
@@ -1987,7 +1987,7 @@ void c_typecheck_baset::typecheck_side_effect_function_call(
         // This is an undeclared function that's not a builtin.
         // Let's just add it.
         // We do a bit of return-type guessing, but just a bit.
-        typet return_type=signed_int_type();
+        typet r = signed_int_type();
```
Comment (peterschrammel) [inline src/ansi-c/c_typecheck_expr.cpp]: `guessed_return_type` ?

File: src/ansi-c/c_typecheck_type.cpp [hunk 3]
```diff
@@ -486,16 +486,16 @@ void c_typecheck_baset::typecheck_code_type(code_typet &type)
   // "A function declarator shall not specify a return type that
   // is a function type or an array type."
 
-  const typet &return_type=follow(type.return_type());
+  const typet &r = follow(type.return_type());
```
Comment (peterschrammel) [inline src/ansi-c/c_typecheck_type.cpp]: `decl_return_type` ?

**Review by peterschrammel — APPROVED** [review]
A few naming suggestions

File: src/ansi-c/c_typecheck_expr.cpp (hunk 1 above)
Comment (tautschnig) [reply src/ansi-c/c_typecheck_expr.cpp]: I have just removed the temporary instead.

File: src/ansi-c/c_typecheck_expr.cpp (hunk 2 above)
Comment (tautschnig) [reply src/ansi-c/c_typecheck_expr.cpp]: Done.

File: src/ansi-c/c_typecheck_type.cpp (hunk 3 above)
Comment (tautschnig) [reply src/ansi-c/c_typecheck_type.cpp]: Thanks, done!

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 028b786937515e7becef04075a6522325bbc94ee).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90980987
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: b490078ef9248318195a6eb5ef5ae2cabf18e22f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90981981

**Review by kroening — APPROVED** [review]

**Pull request merged**